### PR TITLE
🐛 Source Salesforce: fix bug with pagination for BULK API

### LINKED ISSUE #6122: Source-Salesforce: Bulk API limited to 10,000 records
## Enviroment
- **Airbyte version**:  0.29.17-alpha
- **OS Version / Instance**: AWS EC2
- **Deployment**: Docker
- **Source Connector and version**:  salesforce 0.1.0
- **Destination Connector and version**: snowflake 0.3.13
- **Severity**: High
- **Step where error happened**: Sync Job

## Current Behavior
When using the Bulk API option for Salesforce source, it only pulls 10,000 records for any object.
## Expected Behavior
Should be able to pull more than that since the [SF Bulk API documentation](https://developer.salesforce.com/docs/atlas.en-us.salesforce_app_limits_cheatsheet.meta/salesforce_app_limits_cheatsheet/salesforce_app_limits_platform_bulkapi.htm#batch_allocations) does not specify any record limitation, only file size limitation.

I forked the SF repo and changed the limit 
 https://github.com/airbytehq/airbyte/blob/6ceccadf21bd6cbb563b4fb3ed0e7643b209ae88/airbyte-integrations/connectors/source-salesforce/source_salesforce/streams.py#L119 from 10,000 to 500,000 and it works better. Is there any reason this limit was applied in the first place? 

## Are you willing to submit a PR?
Yes

### PR DESCRIPTION
**Pull request by yevhenii-ldv**
## What
*closes #6122.*

## How
*Describe the solution*

## Recommended reading order
1. `x.java`
2. `y.python`

#### Community member or Airbyter
   
- [ ] Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [ ] Secrets in the connector's spec are annotated with `airbyte_secret` 
- [ ] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] Code reviews completed
- [ ] Documentation updated 
    - [ ] Connector's `README.md`
    - [ ] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [ ] Changelog updated in `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)
- [ ] Connector version bumped like described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)
   
#### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items. 
   
- [ ] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] Credentials added to Github CI. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci). 
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing. 
- [ ] New Connector version released on Dockerhub by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)

### CODE REVIEW

**Review by htrueman — APPROVED** [review]

**Review by Zirochkaa — APPROVED** [review]

File: airbyte-integrations/connectors/source-salesforce/source_salesforce/streams.py [hunk 1]
```diff
@@ -186,7 +186,7 @@ def delete_job(self, url: str):
 
     def next_page_token(self, last_record: dict) -> str:
         if self.primary_key and self.name not in UNSUPPORTED_FILTERING_STREAMS:
-            return f"WHERE {self.primary_key} > '{last_record[self.primary_key]}' "
+            return f"WHERE {self.primary_key} >= '{last_record[self.primary_key]}' "
```
Comment (sherifnada) [inline airbyte-integrations/connectors/source-salesforce/source_salesforce/streams.py:189]: are PKs numeric and incremental? why are we not using cursor field here?

File: airbyte-integrations/connectors/source-salesforce/source_salesforce/streams.py [hunk 2]
```diff
@@ -272,11 +272,12 @@ def read_records(
 
             if job_status in ["JobComplete", "Aborted", "Failed"]:
                 self.delete_job(url=job_full_url)
-                pagination_complete = True
+                if job_status in ["Aborted", "Failed"]:
```
Comment (sherifnada) [inline airbyte-integrations/connectors/source-salesforce/source_salesforce/streams.py:275]: should we be raising exception if the job failed?

File: airbyte-integrations/connectors/source-salesforce/source_salesforce/streams.py [hunk 3]
```diff
@@ -66,8 +66,8 @@ def path(self, **kwargs) -> str:
 
     def next_page_token(self, response: requests.Response) -> str:
         response_data = response.json()
-        if len(response_data["records"]) == self.limit and self.primary_key and self.name not in UNSUPPORTED_FILTERING_STREAMS:
-            return f"WHERE {self.primary_key} > '{response_data['records'][-1][self.primary_key]}' "
+        if len(response_data["records"]) == self.page_size and self.primary_key and self.name not in UNSUPPORTED_FILTERING_STREAMS:
+            return f"WHERE {self.primary_key} >= '{response_data['records'][-1][self.primary_key]}' "
```
Comment (sherifnada) [inline airbyte-integrations/connectors/source-salesforce/source_salesforce/streams.py:70]: why aer we comparing PK and not cursor field?

**Review by sherifnada — APPROVED** [review]

File: airbyte-integrations/connectors/source-salesforce/source_salesforce/streams.py (hunk 1 above)
Comment (yevhenii-ldv) [reply airbyte-integrations/connectors/source-salesforce/source_salesforce/streams.py:189]: This is used for `Full Refresh` streams. It turned out that it is impossible to use `Offset `more than `2000` - Salesforce API swears at this, so I had to use the same method as for Incremental, only use `ID`, not `cursor_field`.

File: airbyte-integrations/connectors/source-salesforce/source_salesforce/streams.py (hunk 3 above)
Comment (yevhenii-ldv) [reply airbyte-integrations/connectors/source-salesforce/source_salesforce/streams.py:70]: Answered in the comment above

File: airbyte-integrations/connectors/source-salesforce/source_salesforce/streams.py (hunk 2 above)
Comment (yevhenii-ldv) [reply airbyte-integrations/connectors/source-salesforce/source_salesforce/streams.py:275]: Done

**Pull request merged**